### ISSUE TITLE
Missing gpg-error-config command - build fails

### ISSUE THREAD

**Issue by lfuerderer** [johnschug/rust-gpg-error#7]
When I use [gpgme](https://crates.io/crates/gpgme) as a dependency in my project, its dependency [libgpg-error-sys](https://crates.io/crates/libgpg-error-sys) fails to build.
I get the following error:
```
error: failed to run custom build command for `libgpg-error-sys v0.5.2`

Caused by:
  process didn't exit successfully: `/home/builder/gpg_example/target/debug/build/libgpg-error-sys-25111332c7198a2c/build-script-build` (exit status: 1)
  --- stdout
  scanning file: /home/builder/.cargo/registry/src/github.com-1ecc6299db9ec823/libgpg-error-sys-0.5.2/err-sources.h.in
  scanning file: /home/builder/.cargo/registry/src/github.com-1ecc6299db9ec823/libgpg-error-sys-0.5.2/err-codes.h.in
  scanning file: /home/builder/.cargo/registry/src/github.com-1ecc6299db9ec823/libgpg-error-sys-0.5.2/errnos.in
  cargo:generated=/home/builder/gpg_example/target/debug/build/libgpg-error-sys-85d5fa5da3f84416/out
  cargo:rerun-if-env-changed=LIBGPG_ERROR_PREFIX
  cargo:rerun-if-env-changed=LIBGPG_ERROR_INCLUDE
  cargo:rerun-if-env-changed=LIBGPG_ERROR_LIB_DIR
  cargo:rerun-if-env-changed=LIBGPG_ERROR_LIBS
  cargo:rerun-if-env-changed=LIBGPG_ERROR_CONFIG

  --- stderr
  running: "sh" "-c" "gpg-error-config --version"
  sh: Zeile 1: gpg-error-config: Kommando nicht gefunden.
  command did not execute successfully, got: exit status: 127
  Error: ()
```
As I can see, the command `gpg-error-config` was not found.

I am on an Arch Linux machine with the package [libgpg-error](https://archlinux.org/packages/core/x86_64/libgpg-error/) version 1.46-1 installed.
During debugging, I took the [source code releases of libgpg-error](https://www.gnupg.org/ftp/gcrypt/libgpg-error/) and built them myself.
What I found out:
- Version 1.45 puts the command `gpg-error-config` into the bin directory during the `make install` run.
- Version 1.46 does no longer do this (that's why the command is missing on my system), however there is still a script with this name in the source code at `src/gpg-error-config`.

As a final step, I tried setting the environment variable `LIBGPG_ERROR_CONFIG` to the path of `src/gpg-error-config` in the source code bundle of `libgpg-error` during `cargo build`.
This worked, the rust dependency `libgpg-error-sys` compiles this way.

Honestly, I am not quite sure how to deal with this issue.
It confuses me, that the source code of `libgpg-error` contains a script that is no longer copied to `bin/` during installation.
I don't know if this is a bug in `libgpg-error` or if we are supposed to use another command instead.

**Comment by johnschug** [comment]
It looks like the configuration script (`gpg-error-config`) is [no longer installed by default](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgpg-error.git;a=blob;f=NEWS;h=9818522fa67d97be91656c5798e98bc9952bc348;hb=HEAD) and pkg-config is now the preferred method for configuration. As a potential workaround setting `LIBGPG_ERROR_CONFIG` to `gpgrt-config --libdir=FOO`, where `FOO` is the path that contains your pkgconfig directory, should work for now. 

I've updated the crate on master (and soon crates.io) to use the `system-deps` crate, which uses pkg-config internally, but dependent crates will have to be updated to use the new version as it has breaking changes (new environment variables).

**Comment by alexanderkjall** [comment]
This also affects Debian sid: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1022415

I guess this is a good thread to track for a new release?

**Comment by johnschug** [comment]
I've now published new versions of gpgme (v0.11) and gpg-error (v0.6) that use pkg-config for auto-detection.

**Comment by enhering** [comment]
When trying to compile libgcrypt 1.11.2 with libgpg-error on a custom location, even by setting --with-libgpg-error-prefix to that location on ./configure, I had to do the following to make the compilation work:

cd ${LIBGPGERROR_PATH}/bin
ln -s gpgrt-config gpg-error-config 
cd -

I hope this info helps.
